mongomock/database.py

```diff
@@ -68,10 +72,37 @@ def collection_names(self, include_system_collections=True, session=None):
 
         return self.list_collection_names(session=session)
 
-    def list_collection_names(self, session=None):
+    def _verify_list_collection_supported_op(self, keys):
+        if set(keys) - list_collection_filter_allowed_operators:
+            raise NotImplementedError(
+                'list collection names filter operator {0} is not implemented yet in mongomock '
+                'allowed operators are {1}'.format(keys, list_collection_filter_allowed_operators))
+
+    def list_collection_names(self, filter=None, session=None):
+        """filter: only name field type with eq,ne or regex operator
+
+        session: not supported
+        supported operator are $eq, $ne, $regex
```

Comment:
I suggest we drop that and point to the frozenset instead, otherwise we would need to maintain the list of supported operators in two different places.